src/main/java/com/mineboundteam/minebound/item/tool/MyrialMachete.java

```diff
@@ -0,0 +1,46 @@
+package com.mineboundteam.minebound.item.tool;
+
+import com.mineboundteam.minebound.capabilities.PlayerManaProvider;
+import com.mineboundteam.minebound.magic.OffensiveSpells.TelekineticOffensiveSpell;
+import com.mineboundteam.minebound.magic.SpellItem;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.SwordItem;
+import net.minecraft.world.item.Tier;
+import net.minecraft.world.level.Level;
+
+public class MyrialMachete extends SwordItem {
+
+    private final int manaCost;
+
+    public MyrialMachete(Tier pTier, int pAttackDamageModifier, float pAttackSpeedModifier, Properties pProperties, TelekineticOffensiveSpell.TelekineticOffensiveSpellConfig config) {
+        super(pTier, pAttackDamageModifier, pAttackSpeedModifier, pProperties);
+
+        this.manaCost = config.MANA_COST_PER_HIT.get();
+    }
+
+    @Override
+    public boolean hurtEnemy(ItemStack pStack, LivingEntity pTarget, LivingEntity pAttacker) {
+        if (pAttacker instanceof Player player) {
+            player.getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(mana -> SpellItem.reduceMana(manaCost, player));
+        }
+        return super.hurtEnemy(pStack, pTarget, pAttacker);
+    }
+
+    @Override
+    public void inventoryTick(ItemStack pStack, Level pLevel, Entity pEntity, int pSlotId, boolean pIsSelected) {
+        if (!pLevel.isClientSide() && pEntity instanceof Player player && !pIsSelected) {
+            player.getInventory().removeItem(pStack);
+        }
+    }
+
+    @Override
+    public boolean onDroppedByPlayer(ItemStack item, Player player) {
+        player.getInventory().removeItem(item);
+        return false;
+    }
+
+    // TODO: machete can be placed in storage containers like chests, need to prevent that somehow
```

Comment:
Maybe look into 
```
   public boolean canFitInsideContainerItems() {
      return true;
   }
```

Reply:
Despite what the name implies, the function is only used for shulker boxes and has no effect on traditional storage containers.

Reply:
Could you use inventoryTick to check if the player has a container open and cause it to vanish as soon as there is?

Reply:
Yup, that does work. I don't think it's a perfect solution but it works well enough. Pushed.

Reply:
it's not perfect but I think the alternative would be trying to figure out how to make a custom itemstack or something similar